README.md

```diff
@@ -166,42 +167,75 @@ Alternatively you are able to use any `F[_]` which is convertible to/from `monix
 still can, if you want) and you touch only your business class which is then (de)serialized using provided converter.
 
+Since the API is fully effectful don't forget
+
 The library uses two types of executors - one is for blocking (IO) operations and the second for callbacks. You _have to_ provide both of them:
 1. Blocking executor as `ExecutorService`
 1. Callback executor as `scala.concurrent.ExecutionContext`
 
 ```scala
-import com.typesafe.config.ConfigFactory
-import com.avast.metrics.api.Monitor
-import com.avast.clients.rabbitmq._ // for generic types support
+import java.util.concurrent.{ExecutorService, Executors}
+
 import com.avast.bytes.Bytes
+import com.avast.clients.rabbitmq._
+import com.avast.clients.rabbitmq.api.{Delivery, DeliveryResult, RabbitMQConsumer, RabbitMQProducer}
+import com.avast.metrics.scalaapi.Monitor
+import com.typesafe.config.ConfigFactory
 import monix.eval._
+import monix.execution.Scheduler
+
+import scala.concurrent.duration.Duration
 
 val config = ConfigFactory.load().getConfig("myRabbitConfig")
 
-implicit val ec: ExecutionContext = ???
+implicit val sch: Scheduler = ???
+val monitor: Monitor = ???
+
 val blockingExecutor: ExecutorService = Executors.newCachedThreadPool()
 
-val monitor: Monitor = ???
+// the "FP" way:
+
+val consumerAndProducer: Task[(RabbitMQConsumer[Task], RabbitMQProducer[Task, Bytes])] = RabbitMQConnection
+  .fromConfig[Task](config, blockingExecutor)
+  .flatMap { connection =>
+    /*
+      Here you have created the connection; it's shared for all producers/consumers amongst one RabbitMQ server - they will share a single
+      TCP connection but have separated channels.
+      If you expect very high load, you can use separate connections for each producer/consumer, however it's usually not needed.
+    */
+
+    for {
+      consumer <- connection.newConsumer[Bytes]("consumer", monitor) {
+        case delivery: Delivery.Ok[Bytes] =>
+          println(delivery)
+          Task.now(DeliveryResult.Ack)
+
+        case _: Delivery.MalformedContent =>
+          Task.now(DeliveryResult.Reject)
+      }
 
-// here you create the connection; it's shared for all producers/consumers amongst one RabbitMQ server - they will share a single TCP connection
-// but have separated channels
-// if you expect very high load, you can use separate connections for each producer/consumer, but it's usually not needed
-val rabbitConnection = RabbitMQConnection.fromConfig[Task](config, blockingExecutor) // DefaultRabbitMQConnection[Task]
+      producer <- connection.newProducer("producer", monitor)
+    } yield {
+      (consumer, producer)
+    }
+  }
 
-val consumer = rabbitConnection.newConsumer[Bytes]("consumer", monitor) { 
+// the "common" way:
+
+val rabbitConnection = RabbitMQConnection.fromConfig[Task](config, blockingExecutor).runSyncUnsafe(Duration.Inf) // DefaultRabbitMQConnection[Task]
```

Comment:
`Duration.Inf` is not a good option. You never know what happens, there should always be some timeout.